dvc/command/stages.py

```diff
@@ -0,0 +1,312 @@
+        return desc
+
+    if not stage.deps and not stage.outs:
+        return "No outputs or dependencies."
+
+    if not stage.outs and stage.deps:
+        return "Stage depends on " + part_desc(stage.deps, show_size=False)
+
+    plots_and_metrics = [out for out in stage.outs if out.plot or out.metric]
+    outs = [out for out in stage.outs if not (out.plot or out.metric)]
+
+    outs_desc = ""
+    if outs:
+        outs_desc = "Produces " + part_desc(outs)
+
+    other_outs_desc = ""
+    if plots_and_metrics:
+        other_outs_desc = "Generates " + part_desc(plots_and_metrics, False)
+
+    return ", ".join(filter(None, [outs_desc, other_outs_desc]))
+
+
+def prepare_info(stats: List[Tuple[str, int]]) -> List[Text]:
+    ret = []
+    for key, value in stats:
+        if not value:
+            continue
+        if ret and ret[-1] != PIPE:
+            ret.append(PIPE)
+        ret.append(Text.styled(str(value), style="blue"))
+        ret.append(Text.styled(f" {pluralize(key, value)}", style="dim blue"))
+    return ret
+
+
+def prepare_graph_text(
+    stage: "Stage", graph: "nx.DiGraph" = None
+) -> Optional[Text]:
+    text = None
+    if graph:
+        ancestors = nx.ancestors(graph, stage)
+        fork = Text(FORK, style="bold blue")
+
+        def gen_name(up, down, *args):
+            # if they are both in the same file, let's only show up the names
+            use_name = up.path == down.path and isinstance(up, PipelineStage)
+            return prepare_stage_name(up, link=False, use_name=use_name)
+
+        fragments = []
+        for idx, ancestor in enumerate(ancestors):
+            if ancestor is stage:
+                continue
+            if idx > 0:
+                fragments.append(", ")
+            fragments.append(fork)
+            fragments.append(gen_name(ancestor, stage))
+
+        if fragments:
+            text = Text()
+            for fragment in fragments:
+                text.append(fragment)
+            text.truncate(MAX_TEXT_LENGTH, overflow="ellipsis")
+
+    return text
+
+
+def prepare_stage_name(stage: "Stage", link=False, use_name=False) -> Text:
+    title = Text(overflow="fold")
+    address = stage.addressing
+
+    linked = Style(link="file://" + os.path.abspath(stage.relpath))
+    file_style = Style(color=DVC_BLUE, bold=True, underline=False)
+    name_style = None
+    if link:
+        name_style = linked
+        file_style += linked
+
+    if isinstance(stage, PipelineStage):
+        from_pwd = stage.name == address
+        if from_pwd or use_name:
+            title.append(stage.name, style=name_style)
+        else:
+            title.append(stage.relpath, style=file_style)
+            title.append(f":{stage.name}")
+    else:
+        title.append(address, style=file_style)
+    return title
+
+
+def prepare_description(stage: Stage) -> Union[Markdown, Text]:
+    # Description
+    if stage.desc:
+        # not sure if this was ever intended, but markdown looks nice
```

Comment:
I have raised a question in https://github.com/iterative/dvc/issues/5025#issuecomment-753631328 regarding the intention of `desc`.